rust/src/http2/decompression.rs

```diff
@@ -165,16 +165,16 @@ impl HTTP2DecoderHalf {
         }
     }
 
-    pub fn http2_encoding_fromvec(&mut self, input: &Vec<u8>) {
+    pub fn http2_encoding_fromvec(&mut self, input: &[u8]) {
         //use first encoding...
         if self.encoding == HTTP2ContentEncoding::HTTP2ContentEncodingUnknown {
-            if *input == "gzip".as_bytes().to_vec() {
+            if &input[..] == "gzip".as_bytes() {
```

Comment:
These can now simply be `input == "gzip".as_bytes()`...  Or even `input == b"gzip"`.

Reply:
`input == "gzip".as_bytes()` does not work on rust 1.48 :-/

Reply:
Or do you mean I should invert the commits order..? Testing...